workers/unity/Assets/Playground/Scripts/Worker/SteamClientWorkerConnector.cs

```diff
@@ -0,0 +1,138 @@
+﻿using Improbable.Gdk.Core;
+using Improbable.Worker;
+using UnityEngine;
+
+namespace Playground
+{
+    /// <summary>
+    ///     The SteamClientWorkerConnector class provides a sample solution for connecting a game client published
+    ///     on Steam to a SpatialOS deployment. It provides example code for retrieving a
+    ///     Steam auth session token at runtime using either Steamworks.NET (https://steamworks.github.io) or
+    ///     Facepunch.Steamworks (https://github.com/Facepunch/Facepunch.Steamworks).
+    ///     You can also use the ClientWorkerConnector parent class to parse the Steam auth session token
+    ///     from the commandline arguments in case the Steam auth session token is known in advance.
+    ///     You will need to install either Steamworks.NET (https://steamworks.github.io) or
+    ///     Facepunch.Steamworks (https://github.com/Facepunch/Facepunch.Steamworks) in order to use this code.
+    /// </summary>
+    public class SteamClientWorkerConnector : ClientWorkerConnector
+    {
+        protected override bool ShouldUseLocator()
+        {
+            return true;
+        }
+
+        /// <summary>
+        ///     Creates a Locator configuration where the Steam ticket field is populated through an API call.
+        /// </summary>
+        /// <remarks>
+        ///     All other configuration parameters are populated through command line arguments
+        ///     according to the logic of the parent class implementation. In case a SpatialOS login token is specified,
+        ///     this method will not retrieve a Steam auth session ticket and behave just like the parent class method.
+        ///     In case a Steam auth session ticket is already specified, this method will not retrieve a Steam auth
+        ///     session ticket and behave just like the parent class method (the command line argument takes priority).
+        /// </remarks>
+        /// <param name="workerType">The type of the worker to create.</param>
+        /// <returns>The Locator connection configuration</returns>
+        protected override LocatorConfig GetLocatorConfig(string workerType)
+        {
+            var config = base.GetLocatorConfig(workerType);
+            if (config.LocatorParameters.CredentialsType == LocatorCredentialsType.LoginToken)
+            {
+                // Game client is started via the Launcher. Do not retrieve a Steam auth session ticket and connect normally.
+                return config;
+            }
+
+            if (!string.IsNullOrEmpty(config.LocatorParameters.Steam.Ticket))
```

Comment:
We are still relying on the steam deployment tag to be passed through the command line not be empty. This is now handled by calling config.Validate() in WorkerConnector.cs.